test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+        containerCPUShare: .info.runtimeSpec.linux.resources.cpu.shares,
+		containerAnnotations: .info.runtimeSpec.annotations,
+        containerPodNamespace: .info.runtimeSpec.annotations["io.kubernetes.pod.namespace"],
+        containerPodName: .info.runtimeSpec.annotations["io.kubernetes.pod.name"],
+		containerName: .status.metadata.name,
+		pid: .info.pid, 
+		hostname: .info.runtimeSpec.hostname 
+	}
+	]')
+
+workload_host_cpu_set=$(echo $workload_containers | jq -r '.[]|.pid' | \
+while read i; do taskset -cp $i; done | awk '{printf "{\"hostCPUSet\": \"%s\"}", $6}' | jq -rs)
+
+final_results=$(echo "$workload_containers $workload_host_cpu_set" | jq -rs '.[0] as $file1 | .[1] as $file2 | map($file1[] + $file2[]) | unique')
+
+echo "$final_results"
+`
+
+// Collection data for each container in a node
+type crioContainerData struct {
+	ContainerPodNamespace string            `json:"containerPodNamespace"`
+	ContainerPodName      string            `json:"containerPodName"`
+	ContainerName         string            `json:"containerName"`
+	ContainerCPUSet       string            `json:"containerCPUSet"`
+	ContainerCPUShare     int               `json:"containerCPUShare"`
+	ContainerAnnotations  map[string]string `json:"containerAnnotations"`
+	HostCPUSet            string            `json:"hostCPUSet"`
+	Hostname              string            `json:"hostname"`
+	Pid                   int               `json:"pid"`
+}
+
+// getAnnotationCPUShare Looks through the annotations to find the workload partitioning annotation and
+// returns the cpushares or error if no such annotation exists.
+func (c *crioContainerData) getAnnotationCPUShare() (int, error) {
+	anno := fmt.Sprintf("%s/%s", WorkloadAnnotationPrefix, c.ContainerName)
+	cpuShare, ok := c.ContainerAnnotations[anno]
+	if !ok {
+		return 0, fmt.Errorf("workload annotation of [%s] was expected but not found", anno)
+	}
+
+	// workload cpushare annotations have a json string that will contain `{ "cpushares": <share int> }`
+	// we parse and return that value here.
+	temp := map[string]int{}
+	err := json.Unmarshal([]byte(cpuShare), &temp)
+	if err != nil {
+		return 0, fmt.Errorf("err parsing cpushare json annotation: %w", err)
+	}
+
+	value, ok := temp["cpushares"]
+	if !ok {
+		return 0, fmt.Errorf("err cpushare annotation was incorrect expected format {\"cpushares\": <int> }")
+	}
+
+	return value, nil
+}
+
+var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning node validation", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                      = exutil.NewCLIWithoutNamespace("cpu-partitioning").AsAdmin()
+		ctx                     = context.Background()
+		isClusterCPUPartitioned = false
+	)
+
+	g.BeforeEach(func() {
+		isClusterCPUPartitioned = getCpuPartitionedStatus(oc) == ocpv1.CPUPartitioningAllNodes
+	})
+
+	g.It("crio containers should have correct cpuset and cpushare set", func() {
```

Comment:
nit: more readable `It ("should have correct cpuset and cpushare set in crio containers")`